src/nvim/lua/treesitter.c

```diff
@@ -1032,7 +1037,10 @@ static int node_named_descendant_for_range(lua_State *L)
 
 static int node_next_child(lua_State *L)
 {
-  TSTreeCursor *cursor = treecursor_check(L, lua_upvalueindex(1));
```

Comment:
I'm not sure why this failed. The cursor should always be an upvalue.

Reply:
It doesn't; must have been the `gc` without the early return.